### PR TITLE
Update NewRecordFromFields to not run enumerable twice.

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1141

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/ValueTests.cs [hunk 1]
```diff
@@ -358,6 +359,23 @@ public void NewError()
             Assert.All(combinedError.Errors, (e) => e.Kind = ErrorKind.Custom);
             Assert.Equal("test1", combinedError.Errors.First().Message);
         }
+
+        [Fact]
+        public void NewRecordFromFieldsTest()
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/ValueTests.cs:364]: We should just add a comment like:
// ensure the enumeration is only traversed once.

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/ValueTests.cs:364]: add comment

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**